### DISCUSSION TITLE
Vendor libs like zlib should use distinct types for sizes...

### DISCUSSION THREAD

**Discussion by Beefster09** [odin-lang/Odin#2770] [Proposals]
I hop between Linux and Windows to make sure my game continues to work on both and ran into a discrepancy between platforms because zlib uses differently sized types for compress and decompress buffer sizes depending on whether you build with VC++ or GCC.

I propose that distinct types be used in cases like these to make it harder to accidentally create non-portable code that could otherwise be easily made portable.

**Comment by hasenj** [comment]
You can vary the size of a type without making it distinct. I don't see how _distinct_ helps with making the code portable?

**Reply by Beefster09** [reply]
It causes an error if you use whatever the distinct type is aliasing rather than the alias, so it would force you to use the type alias rather than the underlying type.
